### PR TITLE
Convert duplicate web3 detection from error to warning

### PR DESCRIPTION
**Pull request by danfinlay**
Current behavior will cause uncaught errors on unrelated websites, even if they aren't using web3. Warning is probably the more semantically correct behavior.

### CODE REVIEW

**Comment by Gudahtt** [comment]
>Current behavior will cause uncaught errors on unrelated websites, even if they aren't using web3.

How might this happen? If a site isn't using `web3`, why would that global be defined?

**Comment by Gudahtt** [comment]
Oh I see; this error only applies when another extension is injecting web3 _before_ we are. So this only happens in the case where both extensions are active at once, and they conflict with each other. So the error isn't related to the current site at all, and it'd be thrown regardless of whether they use `web3` or not.

**Review by Gudahtt — COMMENTED** [review]
This does have a functional impact as well; throwing an error here ensures the rest of the script isn't executed. I wondering if this could result in the page being in a broken state, which global objects injected from two different wallets.

**Review by rekmarks — CHANGES_REQUESTED** [review]
We should probably return after warning, and not attempt to initialize our own `window.web3`?

**Comment by Gudahtt** [comment]
Though I do wonder whether we're already leaving things in a broken state in the multi-wallet scenario, given that `window.ethereum` is still being setup regardless.

**Comment by rekmarks** [comment]
Yeah, all bets are off in the multi-wallet scenario, although we can at least take one possible blowup off the table 😛

**Comment by Gudahtt** [comment]
We could (and probably should) check for `window.ethereum` before we set that up too, and that should (I think) have the effect of cleanly disabling MetaMask if another wallet is load first via extension. That can be done in a separate PR though.

**Comment by jacobwillemsma** [comment]
Is this just missing a return after the warn? Continues to show up in my non-web3 website's failures every ~1 week or so.

**Comment by danfinlay** [comment]
Sorry for not following the feedback. I've addressed it now: Detecting another web3 now prevents the rest of our injection logic.

File: app/scripts/inpage.js [hunk 1]
```diff
@@ -65,26 +65,28 @@ initProvider({
...
+  Object.defineProperty(window.ethereum, '_web3Ref', {
+    enumerable: false,
+    writable: true,
+    configurable: true,
+    value: web3.eth,
+  })
+
+  // setup dapp auto reload AND proxy web3
+  setupDappAutoReload(web3, window.ethereum._publicConfigStore)
+}
```
Comment (rekmarks) [inline app/scripts/inpage.js:91]: When this is rebased, the `if` condition should just check if `window.web3` is undefined, and the `if` clause should  `log.warn` that another `window.web3` was detected.

The `else` clause should call `setupWeb3`, per the current state of the file: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/inpage.js

Comment (danfinlay) [reply app/scripts/inpage.js]: Implemented this change but also detect for existing ethereum provider before performing that setup.

**Comment by Gudahtt** [comment]
Does it still make sense to inject `web3` if we're not injecting our provider? :thinking: I'm not sure what the point of that would be.

**Comment by danfinlay** [comment]
> Does it still make sense to inject web3 if we're not injecting our provider?

I agree it is redundant and probably preserves no case, but it does preserve "ease of deprecation removal" for the web3 deprecation, keeping that logic in one place and easy to delete when it's ready to end.

**Comment by danfinlay** [comment]
Opened simplified alternative, since this one crept in scope:
https://github.com/MetaMask/metamask-extension/pull/9832

**Pull request closed without merging**